ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/shared/target/EditRepositoryContainerPage.java

```diff
@@ -0,0 +1,290 @@
+		IDialogSettings settings = getDialogSettings();
+		if (settings != null) {
+			String location = settings.get(SETTINGS_LOCATION_1);
+			if (location != null) {
+				previousLocations.add(location);
+			}
+			location = settings.get(SETTINGS_LOCATION_2);
+			if (location != null) {
+				previousLocations.add(location);
+			}
+			location = settings.get(SETTINGS_LOCATION_3);
+			if (location != null) {
+				previousLocations.add(location);
+			}
+		}
+		return previousLocations.toArray(new String[previousLocations.size()]);
+	}
+
+	@Override
+	public void storeSettings() {
+		String newLocation = furiLocation.getText().trim();
+
+		int length = newLocation.length();
+		if (length > 0 && newLocation.charAt(length - 1) == File.separatorChar) {
+			newLocation = newLocation.substring(0, length - 1);
+		}
+		String[] items = furiLocation.getItems();
+		for (String item : items) {
+			if (item.equals(newLocation)) {
+				// Already have this location stored
+				return;
+			}
+		}
+		IDialogSettings settings = getDialogSettings();
+		if (settings != null) {
+			String location = settings.get(SETTINGS_LOCATION_2);
+			if (location != null) {
+				settings.put(SETTINGS_LOCATION_3, location);
+			}
+			location = settings.get(SETTINGS_LOCATION_1);
+			if (location != null) {
+				settings.put(SETTINGS_LOCATION_2, location);
+			}
+			settings.put(SETTINGS_LOCATION_1, newLocation);
+		}
+	}
+
+	@Override
+	public RepositoryBundleContainer getBundleContainer() {
+		String rawUri = furiLocation != null & !furiLocation.isDisposed() ? furiLocation.getText().trim() : ""; //$NON-NLS-1$
+		return new RepositoryBundleContainer(rawUri, fContainer == null ? List.of() : fContainer.getRequirements());
+	}
+
+	/**
+	 * Validate the input fields before a container is created/edited. The
+	 * page's enablement, message and completion should be updated.
+	 *
+	 * @return whether the finish button should be enabled and container
+	 *         creation should continue
+	 */
+	protected boolean validateInput() {
+		if (furiLocation.isDisposed())
+			return false;
+
+		// Check if the text field is blank
+		if (furiLocation.getText().trim().length() == 0) {
+			setMessage(getDefaultMessage());
+			return false;
+		}
+
+		// Resolve any variables
+		URI location;
+		try {
+			location = RemoteTargetHandle.getEffectiveUri(furiLocation.getText().trim());
+		} catch (CoreException e) {
+			setMessage(e.getMessage(), IMessageProvider.WARNING);
+			return true;
+		} catch (URISyntaxException e) {
+			setMessage(e.getMessage(), IMessageProvider.ERROR);
+			return false;
+		}
+		try {
+			// and be converted to an URL
+			URL url = location.toURL();
+			if ("file".equalsIgnoreCase(url.getProtocol())) { //$NON-NLS-1$
+				File file = new File(location);
+				if (!file.isFile()) {
+					setMessage(NLS.bind(Messages.EditTargetContainerPage_Not_A_File, file.getAbsolutePath()),
+							IMessageProvider.ERROR);
+					return true;
+				}
+			}
```

Comment:
An error is issued but the result is true?

Reply:
This is similar to the other case, e.g. the user might want to point to a file that is (not yet) existing but should still be able to complete the wizard.
I understand that this are maybe strange cases but I don't block the user in such cases from finish the dialog.